pilot/pkg/security/authz/builder/extauthz.go

```diff
@@ -289,6 +289,7 @@ func generateHTTPConfig(hostname, cluster string, status *envoytypev3.HttpStatus
 		},
 		FilterEnabledMetadata: generateFilterMatcher(wellknown.HTTPRoleBasedAccessControl),
 		WithRequestBody:       withBodyRequest(config.IncludeRequestBodyInCheck),
+		ClearRouteCache:       true,
```

Comment:
do we need this conditional? https://github.com/istio/istio/blob/24c6d2151bd1703b76047da97f26f1a8df2edaf2/pilot/pkg/security/authn/v1beta1/policy_applier.go#L163

Reply:
that one is for reqAuthn, I don't think it has anything to do with extAuthz.

Reply:
Doesn't the same apply though? we don't do routing for inbound trafffic for sidecars. Only gw?

Reply:
yeah, so you mean to say we should follow same for external Authz as well? 
Eg. Don't need to clear the cache when it's for sidecars and only clear it when it is for gw

Reply:
I agree, will update this PR to ignore cache clearing for sidecars

Reply:
still working on ^?